### PR TITLE
Basic front-end setup for notifications

### PR DESCRIPTION
**Pull request by rudream**
Basic frontend setup for notifications with some dummy data on how the notifications should be formatted and stored on the backend. 

Currently there is a test button that will trigger a socket event to simulate a notification being received from the backend.

notifications

### CODE REVIEW

File: client/src/components/NavBar.js [hunk 1]
```diff
@@ -77,50 +94,73 @@ const NavBar = ({ history }) => {
...
+      <Drawer
+        variant="temporary"
+        anchor="right"
+        open={navOpen}
+        onClose={() => setNavOpen(false)}
+      >
+        <div className={classes.drawerDiv}>
+          <List component="nav" aria-label="navigation">
+            <ListItem divider className={classes.plateIcon}>
+              <img src={plateLogo} alt="plate icon" />
```
Comment (WhoaNellie) [inline client/src/components/NavBar.js:195]: You could possibly use an [SvgIcon](https://material-ui.com/components/icons/#svgicon) component here

File: client/src/components/NotifsDrawer.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  useEffect(() => {
+    setUnreadCount(notifs.length);
+  }, [notifs, setUnreadCount]);
+
+  const notificationClickHandler = (notif) => {
+    notifs.splice(
+      notifs.findIndex((n) => n._id === notif._id),
+      1
+    );
```
Comment (WhoaNellie) [inline client/src/components/NotifsDrawer.js]: If you pass in the index while mapping `notifs`, you could simplify this to:
```suggestion
  const notificationClickHandler = (notif, index) => {
    notifs.splice( index, 1 );
```

File: client/src/components/NotifsDrawer.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    <Drawer
+      variant="temporary"
+      anchor="right"
+      open={notifsOpen}
+      onClose={() => setNotifsOpen(false)}
+    >
+      <div className={classes.drawerDiv} style={{ width: 350 }}>
+        <List component="nav" aria-label="navigation">
+          <ListItem divider style={{ height: 65, textAlign: "center" }}>
+            <ListItemText primary="NOTIFICATIONS" />
```
Comment (WhoaNellie) [inline client/src/components/NotifsDrawer.js]: For all caps text, it's better to have the literal text be normal case and use a [textTransform](https://www.w3schools.com/cssref/pr_text_text-transform.asp) to make it all-caps. To do it on a ListItemText you could use className or try some [primaryTypographyProps](https://material-ui.com/api/list-item-text/#props)

Just a suggestion, I know this is a lot of trouble for something that won't change it visually at all 🤷‍♀️

File: client/src/components/NotifsDrawer.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+          </ListItem>
+          {notifs.length === 0 && (
+            <ListItem style={{ textAlign: "center", marginTop: "20px" }}>
+              <ListItemText secondary="You have no notifications." />
+            </ListItem>
+          )}
+          {notifs.map((notif) => {
+            if (notif.type === "message") {
```
Comment (WhoaNellie) [inline client/src/components/NotifsDrawer.js]: Could this if statement be wrapped around the ListItem's children? It looks like both messages and other types of notifs use the same ListItem props

File: client/src/notificationsAPI.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      _id: "2",
+      type: "message",
+      name: "Sarah Doe",
+      preview: "That sushi you made yesterday was delicious! Thank you!",
+      link: "/chat/57c86515dgf54510552",
+    },
+    {
+      _id: "3",
+      type: "order",
+      name: "Adam Sandler",
```
Comment (WhoaNellie) [inline client/src/notificationsAPI.js:20]: He is the click handler 
image

File: client/src/components/NavBar.js (hunk 1 above)
Comment (rudream) [reply client/src/components/NavBar.js:195]: Will do, thanks!

File: client/src/components/NotifsDrawer.js (hunk 2 above)
Comment (rudream) [reply client/src/components/NotifsDrawer.js]: Good suggestion, thanks!

File: client/src/components/NotifsDrawer.js (hunk 4 above)
Comment (rudream) [reply client/src/components/NotifsDrawer.js]: Yes I could do that

File: client/src/components/NotifsDrawer.js (hunk 3 above)
Comment (rudream) [reply client/src/components/NotifsDrawer.js]: Very doable 👍

File: client/src/components/NavBar.js (hunk 1 above)
Comment (rudream) [reply client/src/components/NavBar.js:195]: It looks like you cant use a .svg file with SvgIcon

**Pull request merged**